### ISSUE TITLE
Composite chart not clipping children

### ISSUE THREAD

**Issue by Frozenlock** [dc-js/dc.js#1488]
#### Individual charts
The bars and the datapoints are clipped when needed.
![image](https://user-images.githubusercontent.com/1152041/46388826-88f8ba80-c69c-11e8-94be-dbc7c47edd77.png)
![image](https://user-images.githubusercontent.com/1152041/46388832-8d24d800-c69c-11e8-83ef-a707903100bd.png)

#### Composite
![image](https://user-images.githubusercontent.com/1152041/46388808-65ce0b00-c69c-11e8-905c-f9450eaff880.png)

### Potential cause
Wrong `clipPath` referred.
![image](https://user-images.githubusercontent.com/1152041/46389221-bcd4df80-c69e-11e8-9955-9407a7a99bcf.png)

**Comment by gordonwoodhull** [comment]
Took me a while to repro this - turns out it only happens if the composite chart is initialized dynamically with a `div`, instead of being initialized with a selector.

I think the child charts should be referencing the clipPath URL of the parent. This works when the parent is named.

The problem is in `anchorName`:
```js
    _chart.anchorName = function () {
        var a = _chart.anchor();
        if (a && a.id) {
            return a.id;
        }
        if (a && a.replace) {
            return a.replace('#', '');
        }
        return 'dc-chart' + _chart.chartID();
    };
```

In this case anchor `a` is a `div` with no `id`. So it misses the first two cases and instead generates a link to the chartID of the child (which has no clip path).

Would you please try this fix, in `baseMixin.anchor`? I am not sure if it's the cleanest fix but it works for me:

```js
        if (dc.instanceOfChart(parent)) {
            _anchor = parent.anchor();
            if(_anchor.children)
                _anchor = '#' + parent.anchorName();
            _root = parent.root();
            _isChild = true;
        } else if (parent) {
```

The idea is that, if we detect that `parent.anchor()` returns a div, we instead force the parent to generate its `anchorName` and we hold onto this anchor instead.

**Comment by gordonwoodhull** [comment]
Another possibility is to force the composite to have an `id`:

```js
            if(_anchor.children && !_anchor.id)
                _anchor.id = parent.anchorName();
```
![image](https://user-images.githubusercontent.com/1366709/46608489-ec3a8080-cad2-11e8-82ba-c2f5d50fd6af.png)

That's kind of nice and explicit, but maybe kind of rude if the user really doesn't want charts to have ids. (There is no escaping that the clip paths need unique ids.)

**Comment by Frozenlock** [comment]
Eh... I'm an idiot, I pushed the fix to #1491 
However I think it's trivial enough to leave it there.

(You were absolutely right btw, your small change fixed it.)
